crates/network/examples/basic_server.rs

```diff
@@ -0,0 +1,57 @@
+use bevy_app::{App, RunMode, ScheduleRunnerPlugin, Update};
+use bevy_ecs::prelude::{EventReader, Query};
+use rjacraft_network::*;
+use rjacraft_protocol::packets::server::*;
+use serde_json::json;
+
+fn main() {
+    tracing_subscriber::fmt::init();
+    App::new()
+        .add_plugins((
+            NetworkPlugin("0.0.0.0:25565"),
+            ScheduleRunnerPlugin {
+                run_mode: RunMode::Loop { wait: None },
+            },
+        ))
+        .add_systems(Update, (handle_status_requests, handle_disconnect))
+        .run();
+}
+
+fn handle_status_requests(
+    mut connections: Query<&mut RemoteConnection>,
+    mut events: EventReader<ClientStatusRequestEvent>,
+) {
+    for event in events.iter() {
+        if let Ok(mut connection) = connections.get_mut(event.connection) {
+            println!("status for connection: {:?}", connection.remote_addr);
+            let packet = ServerStatusPacket::Response(Response {
```

Comment:
I would like to at leas have a type-safe `Response::new` method which would not need the user to manually construct a valid JSON value. Ideally though, I would prefer a generally type-safe structure.